diff --git a/arch/arm/imx-common/cpu.c b/arch/arm/imx-common/cpu.c
index 5c3c080..fb5995d 100644
--- a/arch/arm/imx-common/cpu.c
+++ b/arch/arm/imx-common/cpu.c
@@ -176,5 +176,8 @@ void arch_preboot_os(void)
 {
 	/* disable video before launching O/S */
 	ipuv3_fb_shutdown();
+
+	struct mxc_ccm_reg *ccm = (struct mxc_ccm_reg *)CCM_BASE_ADDR;
+	setbits_le32(&ccm->analog_pll_video, BM_ANADIG_PLL_VIDEO_POWERDOWN);
 }
 #endif
