diff --git a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
index 780e228..4ab8d64 100644
--- a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
+++ b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
@@ -443,7 +443,6 @@ static int pxp_s_output(struct file *file, void *fh,
 			unsigned int i)
 {
 	struct pxps *pxp = video_get_drvdata(video_devdata(file));
-	struct v4l2_pix_format *fmt = &pxp->fb.fmt;
 	u32 size;
 	int ret, bpp;
 
@@ -451,13 +450,13 @@ static int pxp_s_output(struct file *file, void *fh,
 		return -EINVAL;
 
 	/* Output buffer is same format as fbdev */
-	if (fmt->pixelformat == V4L2_PIX_FMT_RGB24  ||
-		fmt->pixelformat == V4L2_PIX_FMT_YUV32)
+	if (pxp->fb.fmt.pixelformat == V4L2_PIX_FMT_RGB24  ||
+		pxp->fb.fmt.pixelformat == V4L2_PIX_FMT_YUV32)
 		bpp = 4;
 	else
 		bpp = 2;
 
-	size = fmt->width * fmt->height * bpp;
+	size = pxp->fb.fmt.width * pxp->fb.fmt.height * bpp;
 	if (size > pxp->outbuf.size) {
 		if (pxp->outbuf.vaddr)
 			free_dma_buf(pxp, &pxp->outbuf);
@@ -468,9 +467,9 @@ static int pxp_s_output(struct file *file, void *fh,
 	}
 	memset(pxp->outbuf.vaddr, 0x0, pxp->outbuf.size);
 
-	pxp->pxp_conf.out_param.width = fmt->width;
-	pxp->pxp_conf.out_param.height = fmt->height;
-	if (fmt->pixelformat == V4L2_PIX_FMT_RGB24)
+	pxp->pxp_conf.out_param.width = pxp->fb.fmt.width;
+	pxp->pxp_conf.out_param.height = pxp->fb.fmt.height;
+	if (pxp->fb.fmt.pixelformat == V4L2_PIX_FMT_RGB24)
 		pxp->pxp_conf.out_param.pixel_fmt = PXP_PIX_FMT_RGB24;
 	else
 		pxp->pxp_conf.out_param.pixel_fmt = PXP_PIX_FMT_RGB565;
